src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
the code of `dynamic_lib.rs` include libdl for:
 - linux (already defined in rtdeps.rs)
 - android (already defined in rtdeps.rs)
 - macos: *not defined in rtdeps.rs*
 - ios: *not defined in rtdeps.rs*
 - freebsd (libdl is a stub)
 - dragonfly (libdl is a stub)

Could you check if `libdl` is need for use of ld... function under ios and macos ? If yes, it must be defined in `rtdeps.rs`.